src/index.js

```diff
@@ -42,13 +43,40 @@ export default function fetch(url, opts) {
 		const options = getNodeRequestOptions(request);
 
 		const send = (options.protocol === 'https:' ? https : http).request;
+		const { signal } = request;
+		let response = null;
+
+		const abort = ()  => {
+			let error = new AbortError('The user aborted a request.');
+			reject(error);
+			if (request.body && request.body instanceof Stream) {
+				request.body.destroy(error);
```

Comment:
This is only available in node v8.0.0+

Maybe we can just emit an error event in node 4 & 6? Even though this is slightly off-spec